qiita_db/util.py

```diff
@@ -941,20 +1002,21 @@ def get_lat_longs():
     list of [float, float]
         The latitude and longitude for each sample in the database
     """
-    conn = SQLConnectionHandler()
-    sql = """SELECT DISTINCT table_name
-             FROM information_schema.columns
-             WHERE SUBSTR(table_name, 1, 7) = 'sample_'
-                AND table_schema = 'qiita'
-                AND column_name IN ('latitude', 'longitude');"""
-    tables_gen = (t[0] for t in conn.execute_fetchall(sql))
+    with TRN:
+        sql = """SELECT DISTINCT table_name
+                 FROM information_schema.columns
+                 WHERE SUBSTR(table_name, 1, 7) = 'sample_'
+                    AND table_schema = 'qiita'
+                    AND column_name IN ('latitude', 'longitude');"""
+        TRN.add(sql)
+        tables_gen = (t[0] for t in TRN.execute_fetchindex())
```

Comment:
Hmm, shouldn't be considered a blocker comment, though we should keep
this in mind if we keep seeing this in other pull requests.

On (Jul-03-15|17:25), josenavas wrote:
> > @@ -941,20 +1002,21 @@ def get_lat_longs():
> >      list of [float, float]
> >          The latitude and longitude for each sample in the database
> >      """
> > -    conn = SQLConnectionHandler()
> > -    sql = """SELECT DISTINCT table_name
> > -             FROM information_schema.columns
> > -             WHERE SUBSTR(table_name, 1, 7) = 'sample_'
> > -                AND table_schema = 'qiita'
> > -                AND column_name IN ('latitude', 'longitude');"""
> > -    tables_gen = (t[0] for t in conn.execute_fetchall(sql))
> > +    with TRN:
> > +        sql = """SELECT DISTINCT table_name
> > +                 FROM information_schema.columns
> > +                 WHERE SUBSTR(table_name, 1, 7) = 'sample_'
> > +                    AND table_schema = 'qiita'
> > +                    AND column_name IN ('latitude', 'longitude');"""
> > +        TRN.add(sql)
> > +        tables_gen = (t[0] for t in TRN.execute_fetchindex())
> 
> Uhmmm... not sure, we were already doing this before. What others think? I have mixed feelings.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1324/files#r33883432